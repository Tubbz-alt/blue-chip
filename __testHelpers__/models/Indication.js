import BaseModel from "../../src/BaseModel";
import Patient from "./Patient";

export default class Indication extends BaseModel {
  static get hasMany() {
    return [Patient];
  }
}
