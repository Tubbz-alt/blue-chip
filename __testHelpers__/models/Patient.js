import BaseModel from "../../src/BaseModel";
import SettingOfUse from "./SettingOfUse";
import Indication from "./Indication";

export default class Patient extends BaseModel {
  static get belongsTo() {
    return [SettingOfUse, Indication];
  }
}
