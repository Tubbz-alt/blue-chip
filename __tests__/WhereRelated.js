import resources from "../__testHelpers__/fixtures/patientsAndSettingsOfUseNormalized";
import resourcesNoRelationshipData from "../__testHelpers__/fixtures/patientsNoRelationshipDataNormalized";
import Patient from "../__testHelpers__/models/Patient";
import Indication from "../__testHelpers__/models/Indication";

describe("Where related tests", () => {
  test("Single id", () => {
    const patients = Patient.query(resources)
      .includes(["indication"])
      .where({myPatient: true})
      .whereRelated(Indication, {id: "1"})
      .toObjects();
    expect(patients).toMatchSnapshot();
  });
  test("Multiple ids", () => {
    const patients = Patient.query(resources)
      .includes(["indication"])
      .where({myPatient: true})
      .whereRelated(Indication, {id: ["1", "2"]})
      .toObjects();
    expect(patients).toMatchSnapshot();
  });
  test("Should return empty array for unintialized relationships queries", () => {
    const patients = Patient.query(resourcesNoRelationshipData)
      .includes(["indication"])
      .where({myPatient: true})
      .whereRelated(Indication, {id: "1"})
      .toObjects();
    expect(patients).toMatchSnapshot();
  });
});
