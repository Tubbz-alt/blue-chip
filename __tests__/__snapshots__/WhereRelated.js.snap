// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Where related tests Single id 1`] = `
Array [
  Object {
    "createdAt": "2018-10-08T18:39:33.000Z",
    "customers": Array [
      "0000",
    ],
    "id": "667",
    "lastSessionDate": "2016-10-01T00:00:00.000Z",
    "myPatient": true,
    "patientId": "0000",
    "serialNumbers": Array [
      "21601009",
    ],
    "sessionsCount": 127001,
    "indication": Object {
      "id": "1",
      "name": "SCI",
    },
    "steps": 24099980,
    "upTime": 68599053,
  },
]
`;

exports[`Where related tests Multiple ids 1`] = `
Array [
  Object {
    "createdAt": "2018-10-08T18:39:33.000Z",
    "customers": Array [
      "0000",
    ],
    "id": "667",
    "lastSessionDate": "2016-10-01T00:00:00.000Z",
    "myPatient": true,
    "patientId": "0000",
    "serialNumbers": Array [
      "21601009",
    ],
    "sessionsCount": 127001,
    "indication": Object {
      "id": "1",
      "name": "SCI",
    },
    "steps": 24099980,
    "upTime": 68599053,
  },
  Object {
    "createdAt": "2018-10-08T18:39:33.000Z",
    "customers": Array [
      "0001",
    ],
    "id": "668",
    "lastSessionDate": "2016-10-01T00:00:00.000Z",
    "myPatient": true,
    "patientId": "0001",
    "serialNumbers": Array [
      "21601009",
    ],
    "sessionsCount": 127001,
    "indication": Object {
      "id": "2",
      "name": "Stroke",
    },
    "steps": 24099980,
    "upTime": 68599053,
  },
]
`;
